src/lib/nlcManager.js

```diff
@@ -142,47 +192,38 @@ NLCManager.prototype._startTraining = function(){
 		if (this.classifierTraining) {
 			resolve(this.classifierTraining);
 		}
+		// Training with data initialized in opts.
 		else if (this.opts.training_data) {
 			let params = {
 				language: this.opts.classifierLanguage,
 				name: this.opts.classifierName,
 				training_data: this.opts.training_data
 			};
 
-			this.nlc.create(params, (err, response) => {
-				if (err) {
-					reject('Error creating classifier from provided training_data:' + JSON.stringify(err, null, 2));
-				}
-				else {
-					this.classifierTraining = response;
-					resolve(response);
-				}
+			this._createClassifier(params).then((result) => {
```

Comment:
There needs to be a catch clause when calling _createClassifier, which should reject the outer most promise with the caught error.